parsing/pprintast.ml

```diff
@@ -496,18 +496,18 @@ and sugar_expr ctxt f e =
           let print = print_indexop a None assign in
           match path, other_args with
           | Lident "Array", i :: rest ->
-            print ".(" ")" (expression ctxt) [i] rest
+            print ".(" "" ")" (expression ctxt) [i] rest
           | Lident "String", i :: rest ->
-            print ".[" "]" (expression ctxt) [i] rest
+            print ".[" "" "]" (expression ctxt) [i] rest
           | Ldot (Lident "Bigarray", "Array1"), i1 :: rest ->
-            print ".{" "}" (simple_expr ctxt) [i1] rest
+            print ".{" "," "}" (simple_expr ctxt) [i1] rest
           | Ldot (Lident "Bigarray", "Array2"), i1 :: i2 :: rest ->
-            print ".{" "}" (simple_expr ctxt) [i1; i2] rest
+            print ".{" "," "}" (simple_expr ctxt) [i1; i2] rest
           | Ldot (Lident "Bigarray", "Array3"), i1 :: i2 :: i3 :: rest ->
-            print ".{" "}" (simple_expr ctxt) [i1; i2; i3] rest
+            print ".{" "," "}" (simple_expr ctxt) [i1; i2; i3] rest
           | Ldot (Lident "Bigarray", "Genarray"),
             {pexp_desc = Pexp_array indexes; pexp_attributes = []} :: rest ->
-              print ".{" "}" (simple_expr ctxt) indexes rest
+              print ".{" "," "}" (simple_expr ctxt) indexes rest
```

Comment:
I don't think so, since this is the existing `a.{a,b,c,d,e,..}` syntax for `Bigarray.Genarray...` .

Reply:
Ah, fair enough. I wonder if we couldn't deprecate that in favor of the ";" version, for consistency's sake. But that's for later ...

Reply:
i noticed that in the second code example of the top post of this pr, there is a typo where the genarray syntax wrongly uses `;`. just to avoid (unlikely) confusion.

Reply:
Fixed, thanks.

Reply:
Tuples are quite useful  as indices for fixed dimension multidimensional arrays for instance. I think it is worthwhile to keep both options. Moreover, using semicolon `;` for varying length multi-indices and `,` for static length  seems more in tune with the rest of the language — except for the Bigarray generic array sugar. (I have fixed this last typo, thanks).

Reply:
i agree that `;` for variable length collections seems more consistent with the rest of the language. so having both options seems the best solution then.